sycl/include/CL/sycl/detail/image_accessor_util.hpp

```diff
@@ -855,6 +860,88 @@ DataTy ReadPixelDataNearestFiltMode(cl_int4 PixelCoord, id<3> ImgPitch,
   return Color;
 }
 
+// Checks if the PixelCoord is out-of-range, and returns appropriate border or
+// color value at the PixelCoord.
+template <typename DataT>
+DataT getColor(cl_int4 PixelCoord, addressing_mode SmplAddrMode,
+               range<3> ImgRange, id<3> ImgPitch,
+               image_channel_type ImgChannelType,
+               image_channel_order ImgChannelOrder, void *BasePtr,
+               uint8_t ElementSize) {
+  DataT RetData;
+  if (isOutOfRange(PixelCoord, SmplAddrMode, ImgRange)) {
+    cl_float4 BorderColor = (getBorderColor(ImgChannelOrder));
+    RetData = BorderColor.convert<typename TryToGetElementType<DataT>::type>();
+  } else {
+    RetData = ReadPixelData<DataT>(PixelCoord, ImgPitch, ImgChannelType,
+                                   ImgChannelOrder, BasePtr, ElementSize);
+  }
+  return RetData;
+}
+
+// Computes and returns color value with Linear Filter Mode.
+// Steps:
+// 1. Computes the 8 coordinates using all combinations of i0/i1,j0/j1,k0/k1.
+// 2. Calls getColor() on each Coordinate.(Ci*j*k*)
+// 3. Computes the return Color Value using a,b,c and the Color values. 
+template <typename DataT>
+DataT ReadPixelDataLinearFiltMode(cl_int8 CoordValues, cl_float4 abc,
```

Comment:
I see that some values should be with the const modifier. Please add it.

Reply:
done.